django/utils/datastructures.py

```diff
@@ -280,3 +281,86 @@ def __getitem__(self, key):
         if use_func:
             return self.func(value)
         return value
+
+
+def _destruct_iterable_mapping_values(data):
+    for i, elem in enumerate(data):
+        if len(elem) != 2:
```

Comment:
I believe, it can be avoided by explicit unpacking in `for`:
```
for i, (key, value) in enumerate(data):
    if not isinstance(key, str):
        raise ValueError('Element key invalid, only strings are allowed')
    yield (key, value)
```

The error message and behavior is the same.